### PR TITLE
Icoda 112

### PR DESCRIPTION
**Pull request by samhPA**
adding a service to icoda code base.
The service gets dataset from aridhia(calling to aridhia API) and update/insert to icoda database.

The service can be called via our v1/api

### CODE REVIEW

File: src/services/aridhia/aridhia.controller.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	
+			// take each dataset model. if its already in the DB, update the DB. if its not --> insert to the DB
+			for (const model of models) {
+				let ds = new Dataset(model);
+				res = await this.datasetService.replaceOrUpdateOne(ds, {"pid": ds.pid, "activeflag": "active"});
+			}
+	
+			return res;
+
+		} catch (err) {
```
Comment (samhPA) [inline src/services/aridhia/aridhia.controller.js:42]: Add logger and send it to sentry

File: src/services/aridhia/aridhia.service.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+		const res = await this.http.get(this.config.endpoint);
+		return res;
+	}
+
+	resToDataset(res) {
+
+		res = res.data;
+	
+		let doc = {
+				pid: `test-${res.code}`,
```
Comment (samhPA) [inline src/services/aridhia/aridhia.service.js:23]: please change to 'fair-...'

File: src/services/aridhia/aridhia.service.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+				activeflag: "active",
+				contactPoint: res.catalogue.contactPoint,
+				created_at: res.created_at,
+				creator: res.catalogue.creator,
+				datasetfields: { publisher: 'ICODA', phenotypes: [] },
```
Comment (samhPA) [inline src/services/aridhia/aridhia.service.js:28]: fix: Icoda accreditation need to get 'ICODA accreditation'

File: src/resources/dataset/v1/dataset.route.js [hunk 4]
```diff
@@ -16,6 +17,19 @@ const datasetLimiter = rateLimit({
...
+
+router.post('/aridhia', async (req, res) => {
+	
+	// check for a key. return 401, since user unauthorised (caching key is missing or not matched)
+	let parsedBody = req.header('content-type') === 'application/json' ?  req.body : JSON.parse(req.body);
+	if (parsedBody.key !== process.env.cachingkey)
+		return res.status(401).json({ success: false, error: 'Caching could not be started.' });
+		
+	const ac = new AridhiaController();
+	const result = await ac.main();
```
Comment (samhPA) [inline src/resources/dataset/v1/dataset.route.js:29]: remove the await, and make the relevant changes

**Review by RobinKavPA — APPROVED** [review]

**Review by PaulMcCaffertyPA — APPROVED** [review]

**Comment by samhPA** [comment]
Went through a PR meeting and reviewed by RobinKavPA and PaulMcCaffertyPA.

**Pull request merged**